pkg/numaresourcesscheduler/objectstate/sched/sched.go

```diff
@@ -220,6 +226,10 @@ func UpdateSchedulerName(cm *corev1.ConfigMap, name string) error {
 
 }
 
+func GetConfigMapHash(cm *corev1.ConfigMap) string {
+	return fmt.Sprintf("SHA256:%x", sha256.Sum256([]byte(cm.String())))
```

Comment:
so we want hash ALL the objects, metadata included (and not the data), right?

Reply:
Indeed. any change to the ConfigMap will trigger a restart to the pod. 
Also from a computational perspective, it's much easier. IOW we don't need to invent our own function to return a string from a subset of fields in the ConfigMap.

Reply:
ok, works for me

Reply:
On second thought, I'm not sure this is such a good idea. why is a new label to the ConfigMap supposed to restart the scheduler pod? 
WDYT?

Reply:
well, this is the reason I was inquiring. My initial idea was we should only hash the configmap `Spec`, and not the metadata. But I didn't spent too much time thinking about the scenario, so I'm open to suggestions (and not _requiring_ to just hash the `Spec`)

Reply:
let me reevaluate. I inclined to go with hash the ```Spec``` but let's see the cost of such thing